File: 5nthnode.c
int getKthFromLast(struct Node* head, int k) {
    
    struct Node* slow = head;
    struct Node* fast = head;

   
    for (int i = 0; i < k; i++) {
        if (fast == NULL) {
            return -1;  
        }
        fast = fast->next;
    }

    
    while (fast != NULL) {
        slow = slow->next;
        fast = fast->next;
    }

  
    if (slow != NULL) {
        return slow->data;
    }
    return -1;  // Fallback (shouldn’t normally happen)
}

File: 5nthnode.py
class Solution:
    def getKthFromLast(self, head, k):
        
        slow = head
        fast = head

        
        for _ in range(k):
            if fast is None:  
                return -1
            fast = fast.next

        
        while fast is not None:
            slow = slow.next
            fast = fast.next

        #slow now points to the kth node from the end
        return slow.data if slow is not None else -1